deeppavlov/models/go_bot/trippy_preprocessing.py

```diff
@@ -0,0 +1,970 @@
+    text = re.sub("steak house", "steakhouse", text)
+    text = re.sub("turkiesh", " turkish", text)
+    text = re.sub("asian ori$", "asian oriental", text) # Cannot remove the space cuz else it will mess with normal asian ori; probably need some more regex
+    text = re.sub("bristish", " british", text)
+    text = re.sub("asian$", "asian oriental", text)
+    text = re.sub("portugeuse", "portuguese", text)
+    text = re.sub("sea food", "seafood", text)
+    text = re.sub("australian asian", "australasian", text)
+    text = re.sub("^derately", "moderately", text)
+
+
+    return text
+
+
+def tokenize(utt):
+    """
+    Returns tokenized utterance according to TripPy's MultiWoz code
+    """
+    utt_lower = convert_to_unicode(utt).lower()
+    utt_lower = normalize_text(utt_lower)
+    utt_tok = [tok for tok in map(str.strip, re.split("(\W+)", utt_lower)) if len(tok) > 0]
+    return utt_tok
+
+def levenshtein_distance_gen(str1, str2):
+    """
+    Returns Levenshtein distance of two strings
+    """
+    counter = {"+": 0, "-": 0}
+    for edit_code, *_ in ndiff(str1, str2):
+        if edit_code == " ":
+            yield max(counter.values())
+            counter = {"+": 0, "-": 0}
+        else: 
+            counter[edit_code] += 1
+    yield max(counter.values())
+
+
+def get_sys_inform(response, slot_type):
+    """
+    DSTC2-specific function to check if the system informs in current round about specific slot. 
+
+    Args:
+        response: Response data
+        slot_type: What slot type to check for
+    Returns:
+        boolean - if the system informs in round
+    """
+
+    return "inform_" + slot_type in response["act"] if (response is not None) and (response["act"] is not None) else False
+
+
+def get_dialogue_state_sv_dict(context):
+    """
+    Creates dialogue state Slot-Value dict for current context
+    In Trippy, DS is generally given and then converted to an SV dict - We create the DS & convert it in one
+    In order to get an overall diag state we then merge the diag states with more recent ones taking preference
+    """
+    sv_dict = {}
+    # There may be no slots & slot values, though it doesnt really make sense to use TripPy then
+    if "intents" in context:
+        for intent in context["intents"]:
+            if (intent["slots"] is not None) and (len(intent["slots"]) > 0):
+                # Only inform since we only want slots where we have the value (in request we dont)
+                if intent["act"] == "inform":
+                    slot = intent["slots"][0][0]
+                    value = intent["slots"][0][1]
+                    sv_dict[slot] = value
+    return sv_dict
+
+
+def get_tok_label(prev_ds_dict, cur_ds_dict, slot_type, sys_utt_tok,
+                  sys_slot_label, usr_utt_tok, usr_slot_label, dial_id,
+                  turn_id, slot_last_occurrence=True):
+    """
+    Creates labels of 11111111 for where an slot value is uttered by user/system
```

Comment:
Multi Hot Encoding, I went to French restaurant -> 0 0 0 1 0

Reply:
Added explanations to the docstring👍